core/src/main/scala-2.12-/cats/instances/stream.scala

```diff
@@ -6,11 +6,9 @@ import cats.syntax.show._
 import scala.annotation.tailrec
 
 /**
- * For cross compile with backward compatibility
+ * Needed only to avoid some version-specific code in `cats.instances.all`.
  */
-trait LazyListInstances extends StreamInstances with StreamInstancesBinCompat0 {
-  val catsStdInstancesForLazyList = catsStdInstancesForStream
-}
+private[instances] trait LazyListInstances
```

Comment:
Is this still needed?

Reply:
If we don't want two copies of `cats/instances/all.scala` we do need it.

(I guess you could argue that we have two copies of everything else at this point so we might as well do it here.)

Reply:
Yes, but the definition of `AllInstances` in `all.scala` also needs to bring in the `LazyList` instances on 2.13.

Reply:
Ah i see. I didn't see you were able to deprecate all the methods in `StreamInstance` to avoid deprecating the trait. 
Since it confused me a bit, I think we might just as well create two copies for it so that hopefully we can get rid off all `Stream/LazyList` related `ScalaVersionSpecific` stuff in non test code?

Reply:
Sure, I can go ahead and do this.